### PR TITLE
Improve Multi-AZ cluster detection

### PR DESCRIPTION
**Pull request by sergiyvamz**
### Summary

Improve Multi-AZ cluster detection. Increase host-to-dialect cache expiration time.
Related to https://github.com/awslabs/aws-advanced-jdbc-wrapper/discussions/796

### Additional Reviewers

@karenc-bq 
@brunos-bq 
@crystall-bitquill 
@aaronchung-bitquill 

### By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMultiAzDbClusterMysqlDialect.java [hunk 1]
```diff
@@ -31,6 +31,10 @@ public class RdsMultiAzDbClusterMysqlDialect extends MysqlDialect {
 
   private static final String TOPOLOGY_QUERY = "SELECT id, endpoint, port FROM mysql.rds_topology";
 
+  private static final String TOPOLOGY_FUNC_EXIST_QUERY =
```
Comment (brunos-bq) [inline wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMultiAzDbClusterMysqlDialect.java:34]: TOPOLOGY_TABLE_EXIST_QUERY

File: wrapper/src/main/java/software/amazon/jdbc/dialect/DialectManager.java [hunk 2]
```diff
@@ -66,7 +66,7 @@ public class DialectManager implements DialectProvider {
         }
       };
 
-  protected static final long ENDPOINT_CACHE_EXPIRATION = TimeUnit.MINUTES.toNanos(30);
+  protected static final long ENDPOINT_CACHE_EXPIRATION = TimeUnit.HOURS.toNanos(24);
```
Comment (brunos-bq) [inline wrapper/src/main/java/software/amazon/jdbc/dialect/DialectManager.java:69]: NIT: let's add a comment to explain what this value represents
"this value indicates how long... etc etc"

File: wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMultiAzDbClusterMysqlDialect.java [hunk 3]
```diff
@@ -44,8 +48,18 @@ public boolean isDialect(final Connection connection) {
     ResultSet rs = null;
     try {
       stmt = connection.createStatement();
-      rs = stmt.executeQuery(TOPOLOGY_QUERY);
-      return rs.next();
+      rs = stmt.executeQuery(TOPOLOGY_FUNC_EXIST_QUERY);
+
+      if (rs.next()) {
```
Comment (brunos-bq) [inline wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMultiAzDbClusterMysqlDialect.java:53]: NIT: is there any difference in doing
```
if (!rs.next()) {
  return false;
}
rs.close(); stmt.close();
execute query
etc...
```

**Review by brunos-bq — APPROVED** [review]

File: wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMultiAzDbClusterMysqlDialect.java (hunk 3 above)
Comment (sergiyvamz) [reply wrapper/src/main/java/software/amazon/jdbc/dialect/RdsMultiAzDbClusterMysqlDialect.java:53]: No, I don't think there's any difference.

**Pull request merged**